aten/src/ATen/native/sparse/SparseBlas.cpp

```diff
@@ -25,14 +25,24 @@
 namespace at {
 namespace native {
 
-Tensor& addmv_out_sparse_csr(
+Tensor& addmv_out_sparse_compressed(
     const Tensor& self,
     const Tensor& mat,
     const Tensor& vec,
     const Scalar& beta,
     const Scalar& alpha,
     Tensor& result) {
-  TORCH_INTERNAL_ASSERT_DEBUG_ONLY(mat.layout() == kSparseCsr || mat.layout() == kSparseBsr);
+  TORCH_CHECK(
+      mat.layout() != kSparseBsc,
+      "addmv_out_sparse_csr does not support layout SparseBsc, but mat has layout ",
+      mat.layout(),
+      ".");
+  if (mat.layout() == kSparseCsc) {
+    return addmv_out_sparse_compressed(
+        self, mat.to_sparse_csr(), vec, beta, alpha, result);
```

Comment:
This contains an unnecessaily expensive operation (explicit CSC transpose) that can be avoided by generalizing `addmv_out_sparse_csr` to support CSC input. Could be an optimization follow-up.